src/components/users/account/UsernameChangeForm.tsx

```diff
@@ -149,9 +150,7 @@ export const UsernameChangeForm: React.FC = () => {
         ? (<h1>Create a username</h1>)
         : (
           <>
-            <Link href={`/u/${username}`}>
-              <a className='link flex gap-2 items-center'><ArrowLeftCircleIcon className='w-5 h-5' />Back to profile</a>
-            </Link>
+            <a className='link flex gap-2 items-center' href={`/u/${username}`}><ArrowLeftCircleIcon className='w-5 h-5' />Back to profile</a>
```

Comment:
Switch to regular `a` tag instead Next.js `Link`.  Using `Link` will cause Next to prefetch the new profile page `/u/<new_username` as a user typing a new name.